io/plugins/it.geosolutions.hale.io.appschema/src/it/geosolutions/hale/io/appschema/writer/AppSchemaMappingGenerator.java

```diff
@@ -526,10 +527,48 @@ static QName tryInferNamespacePrefix(PropertyDefinition child) {
 						prefix = defaultType.getName().getPrefix();
 				}
 			}
+			else {
+				String newPrefix = getPrefix(dpd, namespaceURI);
+				if (StringUtils.isNotBlank(newPrefix)) {
+					prefix = newPrefix;
+				}
+			}
 		}
 		return new QName(namespaceURI, "", prefix);
 	}
 
+	/**
+	 * Tries to extract the namespace prefix from the provided
+	 * {@link PropertyDefinition} parameter, traversing over the nested complex
+	 * structure of XML definitions.
+	 * 
+	 * @param propertyDefinition the property definition
+	 * @param namespaceURI the expected namespace URI
+	 * @return the extracted namespace prefix if found, or an empty string if
+	 *         not found
+	 */
+	private static String getPrefix(PropertyDefinition propertyDefinition, String namespaceURI) {
+		if (propertyDefinition instanceof AbstractPropertyDecorator) {
+			AbstractPropertyDecorator cdp = (AbstractPropertyDecorator) propertyDefinition;
+			PropertyDefinition property = cdp.getDecoratedProperty();
+			if (property instanceof AbstractPropertyDecorator) {
+				DefinitionGroup declaringGroup = ((AbstractPropertyDecorator) property)
+						.getDeclaringGroup();
+				if (declaringGroup instanceof XmlTypeDefinition) {
+					XmlTypeDefinition typeDef = (XmlTypeDefinition) declaringGroup;
+					// if the type definition namespace URI matches with the
+					// parameter, return the prefix
+					if (typeDef.getName() != null
+							&& namespaceURI.equals(typeDef.getName().getNamespaceURI())) {
+						return typeDef.getName().getPrefix();
+					}
+				}
+			}
+		}
+		// no prefix found, return an empty string
+		return "";
```

Comment:
Are you 100% sure about this? According to @fernandor777 yes this is the expected value because this is an internal. 

In this case it iwll be better to return null.